warehouse/jobs/http_test.go

```diff
@@ -0,0 +1,314 @@
+		Handle:          pgResource.DB,
+		MigrationsTable: "wh_schema_migrations",
+	}).Migrate("warehouse")
+	require.NoError(t, err)
+
+	db := sqlmiddleware.New(pgResource.DB)
+
+	notifier, err := pgnotifier.New(workspaceIdentifier, pgResource.DBDsn)
+	require.NoError(t, err)
+
+	ctx := context.Background()
+
+	now := time.Now().Truncate(time.Second).UTC()
+
+	uploadsRepo := repo.NewUploads(db, repo.WithNow(func() time.Time {
+		return now
+	}))
+	tableUploadsRepo := repo.NewTableUploads(db, repo.WithNow(func() time.Time {
+		return now
+	}))
+	stagingRepo := repo.NewStagingFiles(db, repo.WithNow(func() time.Time {
+		return now
+	}))
+
+	stagingFile := model.StagingFile{
+		WorkspaceID:           workspaceID,
+		Location:              "s3://bucket/path/to/file",
+		SourceID:              sourceID,
+		DestinationID:         destinationID,
+		Status:                whutils.StagingFileWaitingState,
+		Error:                 fmt.Errorf("dummy error"),
+		FirstEventAt:          now.Add(time.Second),
+		UseRudderStorage:      true,
+		DestinationRevisionID: "destination_revision_id",
+		TotalEvents:           100,
+		SourceTaskRunID:       sourceTaskRunID,
+		SourceJobID:           sourceJobID,
+		SourceJobRunID:        sourceJobRunID,
+		TimeWindow:            time.Date(1993, 8, 1, 3, 0, 0, 0, time.UTC),
+	}.WithSchema([]byte(`{"type": "object"}`))
+
+	stagingID, err := stagingRepo.Insert(ctx, &stagingFile)
+	require.NoError(t, err)
+
+	uploadID, err := uploadsRepo.CreateWithStagingFiles(ctx, model.Upload{
+		WorkspaceID:     workspaceID,
+		Namespace:       namespace,
+		SourceID:        sourceID,
+		DestinationID:   destinationID,
+		DestinationType: destinationType,
+		Status:          model.Aborted,
+		SourceJobRunID:  sourceJobRunID,
+		SourceTaskRunID: sourceTaskRunID,
+	}, []*model.StagingFile{{
+		ID:              stagingID,
+		SourceID:        sourceID,
+		DestinationID:   destinationID,
+		SourceJobRunID:  sourceJobRunID,
+		SourceTaskRunID: sourceTaskRunID,
+	}})
+	require.NoError(t, err)
+
+	err = tableUploadsRepo.Insert(ctx, uploadID, []string{
+		"test_table_1",
+		"test_table_2",
+		"test_table_3",
+		"test_table_4",
+		"test_table_5",
+
+		"rudder_discards",
+		"rudder_identity_mappings",
+		"rudder_identity_merge_rules",
+	})
+	require.NoError(t, err)
+
+	t.Run("validate payload", func(t *testing.T) {
+		testCases := []struct {
+			name    string
+			payload StartJobReqPayload
+			isvalid bool
+		}{
+			{
+				name:    "invalid payload",
+				payload: StartJobReqPayload{},
+			},
+			{
+				name: "valid payload",
+				payload: StartJobReqPayload{
+					JobRunID:      "abc",
+					TaskRunID:     "bbc",
+					SourceID:      "cbc",
+					DestinationID: "dbc",
+					WorkspaceID:   "ebc",
+				},
+				isvalid: true,
+			},
+		}
+		for _, tc := range testCases {
+			tc := tc
+
+			t.Run(tc.name, func(t *testing.T) {
+				require.Equal(t, tc.isvalid, validatePayload(&tc.payload))
+			})
+		}
+	})
+
+	t.Run("insert jobs", func(t *testing.T) {
```

Comment:
```suggestion
	t.Run("InsertJobHandler", func(t *testing.T) {
```

We're missing tests for when we might errors from the database (i.e. errors from `a.tableNamesBy` or `a.addJobsToDB`).

Reply:
Missing test case for `StatusJobHandler` too.

Reply:
I am planning to introduce a repo, which will handle `jobStatus`, `tableNamesBy` and `jobStatus` in subsequent PR. I will add a comment for this.